src/Mod/Arch/ArchRoof.py

```diff
@@ -42,241 +52,328 @@ def QT_TRANSLATE_NOOP(ctxt,txt):
+    return getLineIntersections(pt1, pt2, pt3, pt4, infinite1, infinite2)
 
-    '''makeRoof(baseobj,[facenr],[angle],[name]) : Makes a roof based on
-    a closed wire or an object. You can provide a list of angles, run,
-    idrel, thickness, overhang for each edges in the wire to define the
-    roof shape. The default for angle is 45 and the list is
-    automatically complete to match with number of edges in the wire.
-    If the base object is a solid the roof take the shape.'''
 
+def face_from_points(ptLst):
+    ptLst.append(ptLst[0])
+    # Use DraftVecUtils.removeDouble after append as it does not compare the first and last vector:
+    ptLst = DraftVecUtils.removeDoubles(ptLst)
+    ln = len(ptLst)
+    if ln < 4: # at least 4 points are required for 3 edges
+        return None
+    edgeLst = []
+    for i in range(ln - 1):
+        edge = Part.makeLine(ptLst[i], ptLst[i + 1])
+        edgeLst.append(edge)
+    wire = Part.Wire(edgeLst)
+    return Part.Face(wire)
+
+
+def makeRoof(baseobj=None,
+             facenr=0,
+             angles=[45.0], run=[250.0], idrel=[-1], thickness=[50.0], overhang=[100.0],
+             name="Roof"):
+    '''makeRoof(baseobj, [facenr], [angle], [name]): Makes a roof based on
+    a closed wire or an object.
+
+    You can provide a list of angles, run, idrel, thickness, overhang for
+    each edge in the wire to define the roof shape. The default for angle is
+    45 and the list is automatically completed to match the number of edges
+    in the wire.
+
+    If the base object is a solid the roof uses its shape.
+    '''
     if not FreeCAD.ActiveDocument:
         FreeCAD.Console.PrintError("No active document. Aborting\n")
         return
-    import Part
-    obj = FreeCAD.ActiveDocument.addObject("Part::FeaturePython",name)
-    obj.Label = translate("Arch",name)
-    w = None
+    obj = FreeCAD.ActiveDocument.addObject("Part::FeaturePython", name)
+    obj.Label = translate("Arch", name)
+    baseWire = None
     _Roof(obj)
     if FreeCAD.GuiUp:
         _ViewProviderRoof(obj.ViewObject)
     if baseobj:
         obj.Base = baseobj
-        if hasattr(obj.Base,'Shape'):
+        if hasattr(obj.Base, "Shape"):
             if obj.Base.Shape.Solids:
                 if FreeCAD.GuiUp:
                     obj.Base.ViewObject.hide()
             else:
                 if (obj.Base.Shape.Faces and obj.Face):
-                    w = obj.Base.Shape.Faces[obj.Face-1].Wires[0]
+                    baseWire = obj.Base.Shape.Faces[obj.Face-1].Wires[0]
                     if FreeCAD.GuiUp:
                         obj.Base.ViewObject.hide()
                 elif obj.Base.Shape.Wires:
-                    w = obj.Base.Shape.Wires[0]
+                    baseWire = obj.Base.Shape.Wires[0]
                     if FreeCAD.GuiUp:
                         obj.Base.ViewObject.hide()
-        if w:
-            if w.isClosed():
+        if baseWire:
+            if baseWire.isClosed():
```

Comment:
When you have two consecutive ifs, this is the same as an `and` statement. You can do this check on the same line, and then you save one level of indentation.
```py
if `base_wire` and `base_wire.isClosed()`:
    ...
```
The parsing of this expression returns immediately if the first condition is `False`, so it doesn't even test the second one (short-circuit testing).

All over Draft and Arch, there are many such cases of consecutive ifs, making some blocks of code extremely indented and hard to read, so this is something that could be improved.

Reply:
This is part of the old code that I have not modified. Although I did change 'w' to 'baseWire'.
In the case of 2 consecutive ifs the second if is not tested either I think.

Reply:
Yeah, the second `if` is not tested if the first is `False`, but the indentation is still there, which is also important to avoid.